### PR TITLE
Report error message from crash reporter annotations

### PR DESCRIPTION
**Pull request by MatkovIvan**
Report [Application Specific Information](https://developer.apple.com/library/archive/technotes/tn2151/_index.html#//apple_ref/doc/uid/DTS40008184-CH1-APPINFO) from [Apple's crash reporter info](https://opensource.apple.com/source/Libc/Libc-825.26/include/CrashReporterClient.h).

### CODE REVIEW

File: Source/PLCrashLogWriter.m [hunk 1]
```diff
@@ -1111,7 +1132,7 @@ static size_t plcrash_writer_write_mach_signal (plcrash_async_file_t *file, plcr
  * @param file Output file
  * @param siginfo The signal information
  */
-static size_t plcrash_writer_write_signal (plcrash_async_file_t *file, plcrash_log_signal_info_t *siginfo) {
+static size_t plcrash_writer_write_signal (plcrash_async_file_t *file, plcrash_log_signal_info_t *siginfo, struct crashreporter_annotations_t *annotatiions) {
```
Comment (annakocheshkova) [inline Source/PLCrashLogWriter.m]:
```suggestion
static size_t plcrash_writer_write_signal (plcrash_async_file_t *file, plcrash_log_signal_info_t *siginfo, struct crashreporter_annotations_t *annotations) {
```

File: Source/PLCrashLogWriter.m [hunk 2]
```diff
@@ -1156,6 +1177,11 @@ static size_t plcrash_writer_write_signal (plcrash_async_file_t *file, plcrash_l
         rv += plcrash_writer_write_mach_signal(file, siginfo->mach_info);
     }
 
+    /* Message from crash reporter annotations */
+    if (annotatiions != NULL && annotatiions->message != NULL) {
```
Comment (annakocheshkova) [inline Source/PLCrashLogWriter.m]:
```suggestion
    if (annotations != NULL && annotations->message != NULL) {
```

File: Source/PLCrashLogWriter.m [hunk 3]
```diff
@@ -1156,6 +1177,11 @@ static size_t plcrash_writer_write_signal (plcrash_async_file_t *file, plcrash_l
...
+        rv += plcrash_writer_pack(file, PLCRASH_PROTO_SIGNAL_ANNOTATION_MESSAGE_ID, PLPROTOBUF_C_TYPE_STRING, annotatiions->message);
```
Comment (annakocheshkova) [inline Source/PLCrashLogWriter.m]:
```suggestion
        rv += plcrash_writer_pack(file, PLCRASH_PROTO_SIGNAL_ANNOTATION_MESSAGE_ID, PLPROTOBUF_C_TYPE_STRING, annotations->message);
```

File: Source/PLCrashLogWriter.m [hunk 4]
```diff
@@ -1358,6 +1384,18 @@ plcrash_error_t plcrash_log_writer_write (plcrash_log_writer_t *writer,
         plcrash_writer_write_binary_image(file, &image->macho_image);
     }
 
+    /* Try to read crash reporter annotations */
+    struct crashreporter_annotations_t *annotations = NULL;
+    image = plcrash_async_image_containing_address(image_list, siginfo->bsd_info->address);
```
Comment (annakocheshkova) [inline Source/PLCrashLogWriter.m]: Is it redefined? Does it hide some other variable?

File: Source/PLCrashLogWriter.m [hunk 5]
```diff
@@ -1358,6 +1384,18 @@ plcrash_error_t plcrash_log_writer_write (plcrash_log_writer_t *writer,
...
+    if (image != NULL)
+    {
+        plcrash_async_mobject_t mobj;
+        err = plcrash_async_macho_map_section(&image->macho_image, "__DATA", CRASHREPORTER_ANNOTATIONS_SECTION, &mobj);
+        if (err == PLCRASH_ESUCCESS) {
```
Comment (annakocheshkova) [inline Source/PLCrashLogWriter.m]: Nitpick: what is the formatting style here wrt to brackets? Choose one.

Comment (MatkovIvan) [reply Source/PLCrashLogWriter.m]: like in this line. I've already noticed about it myself

File: Source/PLCrashLogWriter.m (hunk 4 above)
Comment (MatkovIvan) [reply Source/PLCrashLogWriter.m]: This var is used inside the loop above. It's safe to just reuse it here

**Comment by MatkovIvan** [comment]
I've noticed that there is a branch introduces custom section which is very similar with it - `feature/PLCR-527-support-plcrash_info`. It's definitely another feature, but we need to check it out before continue work on this. Probably we can pick up something from there.

**Pull request closed without merging**